src/app/acorns/acorns.controller.js

```diff
@@ -5,56 +5,53 @@
-        $scope.checkUser = checkUser;
-
-        checkUser();
-
-        ProfileService.remainingKudos().then(function (val) {
-            Resources.setUserAvailableKudos(val);
-        });
-
-        ProfileService.receivedKudos().then(function (val) {
-            $scope.userReceivedKudos = val;
-            $scope.showLoader = false;
-        });
-
-        $scope.$watch(function () {
-            return Resources.getUserAvailableKudos()
-        }, function (newVal) {
-            if (!isValid(newVal)) $scope.userAvailableKudos = Resources.getUserAvailableKudos();
-        });
-
-        function isValid(value) {
-            return typeof value === "undefined";
-        }
-
-        function checkUser() {
-            ProfileService.checkUser().then(function (val) {
-                val.logged ? $window.location.href = "#/acorns" : $window.location.href = "#/login";
+        'use strict';
+        var AcornController = function ($http, $scope, $window, $cookies, $timeout, $httpParamSerializer, $filter, ProfileService, Challenges, Resources) {
+            $scope.showLoader = true;
+
+            $scope.isValid = isValid;
+            $scope.checkUser = checkUser;
+
+            checkUser();
+
+            function activate() {
+                if ($scope.user == undefined) {
+                    ProfileService.userHome().then(function (user) {
+                        Resources.setCurrentUser(user);
+                        $scope.user = Resources.getCurrentUser();
+                        $scope.user.$$hashKey = "0:0";
+                    });
+                } else {
+                    $scope.user = Resources.getCurrentUser();
+                    $scope.user.$$hashKey = "0:0";
+                }
+            }
+
+            activate();
+
+            $scope.$watch(function () {
+                return Resources.getUserAvailableKudos()
+            }, function (newVal) {
+                if (!isValid(newVal)) $scope.userAvailableKudos = Resources.getUserAvailableKudos();
             });
-        }
-    };
-
-    AcornController.$inject = ['$http', '$scope', '$window', '$cookies', '$timeout', '$httpParamSerializer', '$filter', 'ProfileService', 'Challenges', 'Resources'];
-
-    angular
-        .module('myApp.acorn', [
-            'ngRoute',
-            'ngCookies'
-        ])
-        .controller('AcornController', AcornController);
-})();
\ No newline at end of file
+
+            function isValid(value) {
+                return typeof value === "undefined";
+            }
+
+            function checkUser() {
+                ProfileService.checkUser().then(function (val) {
+                    val.logged ? $window.location.href = "#/acorns" : $window.location.href = "#/login";
+                });
+            }
+        };
+
+        AcornController.$inject = ['$http', '$scope', '$window', '$cookies', '$timeout', '$httpParamSerializer', '$filter', 'ProfileService', 'Challenges', 'Resources'];
+
+        angular
```

Comment:
Rule 1 of the SOLID principles and Rule 1 of JohnPapas  styleguide - Single Responsibility principle. Better to define one component per file. https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#style-y001

Here, you're declaring both the module and the controller in the same file, which might make it harder to find the module. Probably easier to move it out to `acorn.module.js` or something similar, so it's easily findable.